### PR TITLE
DINT-246: Configure alloy with cookie restrictions.

### PR DESCRIPTION
**Pull request by admanesachin**
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: src/alloy.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ const configure = async (): Promise<AlloyInstance> => {
             orgId: aepCtx.imsOrgId as string,
             // TODO ahammond: remove debug when feature complete
             debugEnabled: true,
```
Comment (aniham) [inline src/alloy.ts:22]: i know this isn't part of your ticket but i noticed it and we are so close to release - let's remove the `debugEnabled: true`

File: src/alloy.ts [hunk 2]
```diff
@@ -84,5 +85,25 @@ const hasConfig = (): boolean => {
     );
 };
 
+const setConsent = async (): Promise<void> => {
+    const mg_dnt_cookie = document.cookie.indexOf("mg_dnt") !== -1;
```
Comment (aniham) [inline src/alloy.ts:89]: please rename to `doNotTrackCookie`

File: src/alloy.ts [hunk 3]
```diff
@@ -84,5 +85,25 @@ const hasConfig = (): boolean => {
...
+    const instance = await getAlloy();
+    const consent = mg_dnt_cookie ? "out" : "in";
+
+    instance("setConsent", {
+        consent: [
+            {
+                standard: "Adobe",
+                version: "1.0",
+                value: {
+                    general: consent,
```
Comment (aniham) [inline src/alloy.ts:99]: i'd get rid of the `consent` variable altogether and change this to `general: doNotTrackCookie ? 'out' : 'in'`

File: src/alloy.ts [hunk 4]
```diff
@@ -84,5 +85,25 @@ const hasConfig = (): boolean => {
...
+                },
+            },
+        ],
+    });
+};
+
+setInterval(setConsent, 1000);
```
Comment (aniham) [inline src/alloy.ts:106]: i understand why you're doing this but a first time reader might be confused - could you please add a comment explaining why we're retrying?

**Review by aniham — CHANGES_REQUESTED** [review]
@admanesachin looks great overall!!!!

i had some style etc comments AND also can you please add proper documentation to the readme explaining how the cookie is used, how to set, and so on?

**Comment by aniham** [comment]
@admanesachin also looks like tests and/or linter is failing

you can repro locally by running `npm run test` `npm lint:fix` `npm format:fix` - additionally, can you either add a note to the readme or a git hook for people to do this before submitting PRs - seems to be a common problem when people forget to do this

File: src/alloy.ts (hunk 3 above)
Comment (aniham) [reply src/alloy.ts:99]: @admanesachin i just re-read this - this doesn't meet the requirements that shankar described right?

a shopper is going to be *opted in* by default - we want them to be *opted out* by default

the `dnt` cooking will not work at all in that way. the whole point of it is that it's an opt out cookie, not an opt in cookie. can you start a convo with shankar in the dev channel on this?

File: src/alloy.ts (hunk 1 above)
Comment (admanesachin) [reply src/alloy.ts]: Fixed.

File: src/alloy.ts (hunk 4 above)
Comment (admanesachin) [reply src/alloy.ts:104]: Added comment.

**Review by aniham — APPROVED** [review]

**Pull request merged**